O/OptimPack/build_tarballs.jl

```diff
@@ -0,0 +1,44 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "OptimPack"
+version = v"3.1.0"
+
+# Collection of sources required to build OptimPAck
+sources = [
+    "https://github.com/emmt/OptimPack/releases/download/v$(version)/optimpack-$(version).tar.gz" =>
+    "fa1efdbebf6efa42a1d6b4f6223c7ca0b871fad5c0b53f7dc9be296e4c766190",
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/optimpack-*
+./configure --prefix=$prefix --host=$target --enable-shared=yes --enable-static=no
+FLAGS=()
+if [[ "${target}" == *-mingw* ]]; then
+    # This is needed in order to build the shared library on Windows
+    FLAGS+=(LDFLAGS="-no-undefined")
```

Comment:
I updated the script to make it compatible with `master` version of BinaryBuilder.

When building for Windows, I was getting this error:
```
libtool:   error: can't build i686-w64-mingw32 shared library unless -no-undefined is specified
```
I simply followed the error message and added the `-no-undefined` flag for the linker, but I'm not convinced this is the right solution here :thinking:  At least it makes it possible to build the project

Reply:
Thanks for the fix. I do not understand myself the meaning of this flag. Let us give it a try.